extensions/github1s/src/interfaces/github-api-rest.ts

```diff
@@ -212,3 +220,150 @@ export const getGitHubCommitDetail = (
 		options
 	);
 };
+
+const repoFileTreeCacheKey = (owner: string, repo: string, ref: string) => {
+	return `${owner}/${repo}/${ref}`;
+};
+
+async function readGitHubDirectoryCached(
+	owner: string,
+	repo: string,
+	ref: string,
+	path: string,
+	options?: RequestInit
+): Promise<any | null> {
+	const cacheKey = repoFileTreeCacheKey(owner, repo, ref);
+	if (
+		!repoFileTreeCache.has(cacheKey) ||
+		repoFileTreeCache.get(cacheKey).expireTime < new Date()
+	) {
+		// also populate the cache
+		await getGitHubAllFiles(owner, repo, ref);
```

Comment:
Thank you for this pull request.

In fact we have already cached the file tree here:
https://github.com/conwnet/github1s/blob/HEAD/extensions/github1s/src/providers/fileSearchProvider.ts#L65-L75

We will request both the whole file tree and the shallow file tree at the same time now, because we want to make the workspace avaliable as soon as possible.


You can test current pull request here:
https://github1s-git-fork-bytelegend-cache-file-tree-vscode-github1s.vercel.app/chromium/chromium

The file tree will show only after the recursive request is completed, the recursive request may much slower than the shallow request.

I think we may should use the recursive request to cache the file tree when users not provide a OAuth token. In this way we can reduce the rate limit problem.

Reply:
I see. The change is dedicated for my use case that I want to reduce GH API calls for users behind GFW. It may not be generic enough to go into the upstream. Anyway, I'll close this PR, but still keep this change in my fork. Thanks for the awesome project!